# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [3]:
import numpy as np
from scipy.stats import stats

diet1 = np.array([2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5])
diet2 = np.array([3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2])
mean1 = diet1.mean()
mean2 = diet2.mean()
sdt1 = diet1.std()
sdt2 = diet2.std()

print(mean1, mean2)
print(sdt1, sdt2)

t_value, p_value = stats.ttest_ind(diet1, diet2)

print('Valor t:', t_value)
print('Valor p:', p_value)

2.542857142857143 3.0285714285714285
0.306394436993246 0.13850513878332382
Valor t: -3.5383407969933938
Valor p: 0.004083270191713912


/tmp/ipykernel_683/936727710.py:14: DeprecationWarning: Please import `ttest_ind` from the `scipy.stats` namespace; the `scipy.stats.stats` namespace is deprecated and will be removed in SciPy 2.0.0.
  t_value, p_value = stats.ttest_ind(diet1, diet2)


Para los datos proporcionados, al realizar la prueba obtenemos:

Valor t: -3.5383
Valor p: 0.0040
Podemos concluir que hay evidencia suficiente para rechazar la hipótesis nula. Por lo tanto, podemos afirmar que existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta.

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

In [1]:
import numpy as np
from scipy.stats import f_oneway

# Datos de rendimiento de maíz por fertilizante

fertilizante1 = np.array([20, 21, 20, 19, 20])
fertilizante2 = np.array([22, 21, 23, 22, 21])
fertilizante3 = np.array([24, 23, 22, 23, 24])

# ANOVA

f_statistic, p_value = f_oneway(fertilizante1, fertilizante2, fertilizante3)

print('Estadístico F:', f_statistic)
print('Valor p:', p_value)

Estadístico F: 20.315789473684188
Valor p: 0.00014047824793190475


Como el valor p (0.00014) es menor que un nivel de significancia común como 0.05, podemos concluir que hay evidencia suficiente para rechazar la hipótesis nula de que no hay diferencia en el rendimiento promedio del maíz entre los tres tipos de fertilizantes. Por lo tanto, podemos concluir que existe una diferencia significativa en el rendimiento promedio del maíz entre al menos uno de los tipos de fertilizantes. 

Comparacion entre Fertilizantes Utilizando Turkey HSD

In [3]:
import numpy as np
import pandas as pd
import statsmodels.api as sm
from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Datos de rendimiento de maíz por fertilizante
fertilizante1 = np.array([20, 21, 20, 19, 20])
fertilizante2 = np.array([22, 21, 23, 22, 21])
fertilizante3 = np.array([24, 23, 22, 23, 24])

data = np.concatenate([fertilizante1, fertilizante2, fertilizante3])
labels = ['Fertilizante 1'] * len(fertilizante1) + ['Fertilizante 2'] * len(fertilizante2) + ['Fertilizante 3'] * len(fertilizante3)
df = pd.DataFrame({'Rendimiento': data, 'Fertilizante': labels})

# Realizar Tukey HSD
tukey_result = pairwise_tukeyhsd(df['Rendimiento'], df['Fertilizante'])
print(tukey_result)

       Multiple Comparison of Means - Tukey HSD, FWER=0.05        
    group1         group2     meandiff p-adj  lower  upper  reject
------------------------------------------------------------------
Fertilizante 1 Fertilizante 2      1.8 0.0099 0.4572 3.1428   True
Fertilizante 1 Fertilizante 3      3.2 0.0001 1.8572 4.5428   True
Fertilizante 2 Fertilizante 3      1.4 0.0409 0.0572 2.7428   True
------------------------------------------------------------------


 Los resultados de Tukey HSD sugieren que el Fertilizante 3 es el más efectivo en términos de aumentar el rendimiento de maíz en comparación con los otros dos fertilizantes probados.